Integrations/ExtraHop/ExtraHop.py

```diff
@@ -0,0 +1,385 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+
+import json
+import requests
+from distutils.util import strtobool
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+APIKEY = demisto.params().get('apikey')
+SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else \
+    demisto.params()['url']
+USE_SSL = not demisto.params().get('insecure', False)
+BASE_URL = SERVER + '/api/v1/'
+HEADERS = {
+    'Accept': 'application/json',
+    'Authorization': 'ExtraHop apikey={}'.format(APIKEY)
+}
+if not demisto.params().get('proxy'):
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+# 'response' is a container for paginated results
+response = []
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, url_suffix, data=None, payload=None):
+    data = json.dumps(data)
+    try:
+        res = requests.request(
+            method,
+            BASE_URL + url_suffix,
+            verify=USE_SSL,
+            data=data,
+            headers=HEADERS,
+            params=payload
+        )
+    except requests.exceptions.RequestException:  # This is the correct syntax
+        return_error('Failed to connect to - {} - Please check the URL'.format(BASE_URL))
+    # Handle error responses gracefully
+    if res.status_code == 204:
+        return demisto.results('Successful Modification')
+    elif res.status_code not in {200, 204, 201}:
+        return_error('Error in API call to ExtraHop {} - {}'.format(res.status_code, res.reason))
```

Comment:
please switch the error to state the code and reason explicitly, and also go over the other format strings and use named format strings.(most importantly, the ones that have are than one variable. 
example:
`Error in API call. code:{code}; reason: {reason}'.format(code=res.status_code, reason=res.reason)`